### PR TITLE
(#4578) - Coerce options which should be numbers to number

### PR DESCRIPTION
**Pull request by wheresrhys**
Addresses https://github.com/pouchdb/pouchdb/issues/4578 @daleharvey 

Let me know if I've missed any other points in the code where there should be additional changes to address this

### CODE REVIEW

File: lib/mapreduce/index.js [hunk 1]
```diff
@@ -190,6 +190,38 @@ function addHttpParam(paramName, opts, params, asJson) {
...
+    group_level: coerceInteger(opts.group_level),
+    limit: coerceInteger(opts.limit),
+    skip: coerceInteger(opts.skip)
+  });
+}
+
+function checkPositiveInteger (number) {
+  if (number) {
+    if (typeof number !== 'number') {
+      return  new QueryParseError('Invalid value for integer: "' +
```
Comment (daleharvey) [inline lib/mapreduce/index.js]: tiny nit: extra space `return new QueryParseError`

**Comment by daleharvey** [comment]
@wheresrhys Thanks for the PR, the new testing is great

The tests are failing on this however though, I am assuming from the error that `Object.assign` is going to have spotty support between browsers and node versions. Even fixing that doing whole object clone to work on 3 values seems wrong

```
opts.group_level = ('group_level' in opts) ? coerceInt(opts.group_level) : undefined;
// Same for skip etc
```

or similiar for each of the 3 values seems clearer and much less risky code here. 

Last question would be for coerce, why would a simple `parseInt(number, 10)` not suffice?

**Comment by wheresrhys** [comment]
Object assign is used to avoid mutating the options object. Perhaps that is
a bit too purist as it's only primitive values being altered.

E.g. Parseint('1 acacia road') will return 1, so that's why the extra check
is there. Would seem wrong to allow the above but error on e.g. 1.2

**Comment by daleharvey** [comment]
Ok thanks, `coerce` vs `parseInt` makes sense (it may be useful to add a comment to that function to explain it). 

The code + tests look good for me now, the tests however are failing so will need to take a look at that, if you have problems running the test suite then just ask, thanks a lot

**Comment by wheresrhys** [comment]
I forgot to return the object from coerceOptions. Fixed now

**Comment by wheresrhys** [comment]
I can't recreate that failed build locally ^

**Comment by daleharvey** [comment]
Yup this is looking great now, thanks, merged in https://github.com/pouchdb/pouchdb/commit/112faf9945e8fc746cff20f62df47a5e4f4425a6

**Comment by daleharvey** [comment]
I forgot to mention sorry, our tests arent 100% reliable and occasionally fail with an intermittent, trying to chase them down but restarting the test and it was green so you are all good

**Pull request closed without merging**